pkg/config/config_test.go

```diff
@@ -0,0 +1,210 @@
+package config_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	configpkg "github.com/kairos-io/kcrypt/pkg/config"
+)
+
+var _ = Describe("Config", func() {
+	var tmpDir string
+	var err error
+
+	BeforeEach(func() {
+		tmpDir, err = os.MkdirTemp("", "kcrypt-configuration-*")
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	Describe("GetConfiguration", func() {
+		When("the no relevant block exists", func() {
+			It("returns empty Config", func() {
+				c, err := configpkg.GetConfiguration([]string{tmpDir})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(c.Kcrypt.Server).To(BeEmpty())
+				Expect(c.Kcrypt.UUIDLabelMappings).To(BeEmpty())
+			})
+		})
+
+		When("a kcrypt block exists", func() {
+			var tmpFile *os.File
+
+			BeforeEach(func() {
+				tmpFile, err = os.CreateTemp(tmpDir, "config-*.yaml")
+				Expect(err).ToNot(HaveOccurred())
+				data := []byte(`#cloud-config
+kcrypt:
+  challenger_server: http://test.org:8082
+
+  uuid_label_mappings:
+    COS_PERSISTENT: some_uuid_here
+`)
+				err := os.WriteFile(tmpFile.Name(), data, 0744)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("returns the Config", func() {
+				c, err := configpkg.GetConfiguration([]string{tmpDir})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(c.Kcrypt.Server).To(Equal("http://test.org:8082"))
+				Expect(c.Kcrypt.UUIDLabelMappings["COS_PERSISTENT"]).To(Equal("some_uuid_here"))
+			})
+		})
+
+		When("multiple kcrypt block exist", func() {
+			var tmpFile1, tmpFile2 *os.File
+
+			BeforeEach(func() {
+				tmpFile1, err = os.CreateTemp(tmpDir, "config-*.yaml")
+				Expect(err).ToNot(HaveOccurred())
+				data := []byte(`#cloud-config
+kcrypt:
+  challenger_server: http://test.org:8082
+`)
+				err := os.WriteFile(tmpFile1.Name(), data, 0744)
+				Expect(err).ToNot(HaveOccurred())
+
+				tmpFile2, err = os.CreateTemp(tmpDir, "config-*.yaml")
+				Expect(err).ToNot(HaveOccurred())
+				data = []byte(`#cloud-config
```

Comment:
```suggestion
				data = []byte(`
```

Cosmetics, just to make it more realistic and see if it breaks by avoiding the header